R/derive_params_growth_age.R

```diff
@@ -0,0 +1,265 @@
+#'
+#'
+#' @family der_prm_bds_vs
+#'
+#' @keywords der_prm_bds_vs
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(lubridate)
+#' library(rlang)
+#' library(admiral)
+#'
+#' advs <- dm_peds %>%
+#'   select(USUBJID, BRTHDTC, SEX) %>%
+#'   right_join(., vs_peds, by = "USUBJID") %>%
+#'   filter(USUBJID != "PEDS-1010") %>%
+#'   mutate(
+#'     VSDT = ymd(VSDTC),
+#'     BRTHDT = ymd(BRTHDTC)
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AGECUR_D,
+#'     new_var_unit = CURU_D,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "days",
+#'     trunc_out = FALSE
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AGECUR_M,
+#'     new_var_unit = CURU_M,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "months",
+#'     trunc_out = FALSE
+#'   ) %>%
+#'   mutate(
+#'     AGECUR = ifelse(AGECUR_D >= 365.25 * 2, AGECUR_M, AGECUR_D),
+#'     AGECURU = ifelse(AGECUR_D >= 365.25 * 2, CURU_M, CURU_D)
+#'   )
+#'
+#' #' metadata is in months
+#' cdc_meta_criteria <- admiralpeds::cdc_wtage %>%
+#'   mutate(
+#'     age_unit = "months",
+#'     SEX = ifelse(SEX == 1, "M", "F")
+#'   )
+#'
+#' #' metadata is in days
+#' who_meta_criteria <- bind_rows(
+#'   (admiralpeds::who_wt_for_age_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       age_unit = "days"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_age_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       age_unit = "days"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(AGE = Day)
+#'
+#' criteria <- bind_rows(
+#'   cdc_meta_criteria,
+#'   who_meta_criteria
+#' ) %>%
+#'   rename(AGEU = age_unit)
+#'
+#' derive_params_growth_age(
+#'   advs,
+#'   sex = SEX,
+#'   age = AGECUR,
+#'   age_unit = AGECURU,
+#'   meta_criteria = criteria,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTASDS",
+#'     PARAM = "Weight-for-age z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTAPCTL",
+#'     PARAM = "Weight-for-age percentile"
+#'   )
+#' )
+derive_params_growth_age <- function(dataset,
+                                     sex,
+                                     age,
+                                     age_unit,
+                                     meta_criteria,
+                                     parameter,
+                                     bmi_cdc_correction = FALSE,
+                                     set_values_to_sds = NULL,
+                                     set_values_to_pctl = NULL) {
+  # Apply assertions to each argument to ensure each object is appropriate class
+  # need to add assertion for dataset that checks columns in sex, age, age_unit are present
+  assert_data_frame(meta_criteria, required_vars = exprs(SEX, AGE, AGEU, L, M, S))
+  assert_expr(enexpr(parameter))
+  assert_varval_list(set_values_to_sds, optional = TRUE)
+  assert_varval_list(set_values_to_pctl, optional = TRUE)
+
```

Comment:
I think because the user may have so much freedom to put stuff in these arguments, its hard to check for and catch every case, I think as long as we write thoughtful vignettes and leave enough documentation, there is still some responsibility left to the user/programmer to properly implement and check what theyre feeding in is correct